### PR TITLE
Add typescript support

### PR DESCRIPTION
**Pull request by esdras**

### CODE REVIEW

File: syntax/vue.vim [hunk 1]
```diff
@@ -50,6 +50,12 @@ syntax include @JS syntax/javascript.vim
 unlet! b:current_syntax
 syntax region javascript keepend matchgroup=Delimiter start=/<script\( lang="babel"\)\?\( type="text\/babel"\)\?>/ end="</script>" contains=@JS fold
 
+syntax include @TS syntax/typescript.vim
+if exists("b:current_syntax")
+  unlet b:current_syntax
+endif
+syntax region typescript keepend matchgroup=Delimiter start=/<script\( ts\| lang="ts"\| lang="typescript"\)>/ end="</script>" contains=@TS fold
```
Comment (posva) [inline syntax/vue.vim]: I think it's better to allow any attribute with something like ` \_[^>]*` as in all other script syntaxes

**Comment by esdras** [comment]
I followed the examples, tested, and it is working fine.
Feel free to suggest further changes,  not sure if there is some edge case that I'm not accounting for.

**Comment by posva** [comment]
Thanks a lot! We'll see if we get feedback 🙂

**Comment by esdras** [comment]
You're welcome, thank you for the plugin, great job.
Cheers.

**Pull request merged**